### DISCUSSION TITLE
Java syntax coloring looks wrong

### DISCUSSION THREAD

**Discussion by nickdos** [massCodeIO/massCode#434] [General]
Hi All,

Is the following the expected syntax coloring for Java code?

![image](https://github.com/massCodeIO/massCode/assets/532845/4d2a6ccc-517b-414f-8e87-42335e60fb2e)

It looks buggy to me - weird single letter color changes, and nothing like what VSCode, IntelliJ or TextMate provide.

Using a light theme is even worse as the different red/purple colors are indistinguishable.

![image](https://github.com/massCodeIO/massCode/assets/532845/8176a2a2-a76b-4212-90db-5bd8bd3cddd6)

**Comment by antonreshetov** [comment]
I'm not familiar with Java, but I think you're missing semicolons, which is why the syntax highlighting is broken. Although as I understand in Java semicolons are not required, nevertheless it seems to be the way `.tmLanguage` was described for Java.

**Reply by nickdos** [reply]
Thanks - that would be it. I had this snippet from ages ago and I now recall it was from Grails, which is Groovy based.

The`spock.lang.Specification` triggered me into assuming it was Java.
